### PR TITLE
Show comments in the Timebar

### PR DESCRIPTION
**Pull request by MrPercheul**
Base implementation to have comments shown in the time bar using a modified version of [videojs-markers](http://sampingchuang.com/videojs-markers) (to make it compatible with react). 

Comments are shown as a colored dot on the time bar and expanded on hover.
Upon reaching the comment, it also appears on top of the video in a 0.8 opacity banner

by default disabled in settings.py via TIMESTAMP_IN_TIMEBAR

NOTE : small edit of .pre-commit-config.yaml to have the pre-commit running successfully

### CODE REVIEW

**Comment by MrPercheul** [comment]
For 038dd9e, the checks were not all successfult (pytest issue) BUT by changing the paths from docker-compose-dev.yaml to relative paths, the checks work!

**Comment by mgogoulos** [comment]
Hi, I'm trying to get this working, have set the env as True to settings and then tried to run the frontend build. Here's what I get:

```
docker-compose -f docker-compose-dev.yaml exec -T frontend npm run dist

> mediacms-frontend@0.9.1 dist /home/mediacms.io/mediacms/frontend
> mediacms-scripts rimraf ./dist && mediacms-scripts build --config=./config/mediacms.config.js --env=dist

sh: 1: mediacms-scripts: not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! mediacms-frontend@0.9.1 dist: `mediacms-scripts rimraf ./dist && mediacms-scripts build --config=./config/mediacms.config.js --env=dist`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the mediacms-frontend@0.9.1 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-05-03T18_44_04_160Z-debug.log
```

I am reverting the change you did on docker-compose-dev.yaml, and then rerun this, and this time I get

```
docker-compose -f docker-compose-dev.yaml exec -T frontend npm run dist

> mediacms-frontend@0.9.1 dist /home/mediacms.io/mediacms/frontend
> mediacms-scripts rimraf ./dist && mediacms-scripts build --config=./config/mediacms.config.js --env=dist

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating










Failed to compile. 

resolve 'video.js' in '/home/mediacms.io/mediacms/frontend/src/static/js/components/comments'
  Parsed request is a module
  using description file: /home/mediacms.io/mediacms/frontend/package.json (relative path: ./src/static/js/components/comments)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /home/mediacms.io/mediacms/frontend/src/static/js/components/comments/node_modules doesn't exist or is not a directory
      /home/mediacms.io/mediacms/frontend/src/static/js/components/node_modules doesn't exist or is not a directory
      /home/mediacms.io/mediacms/frontend/src/static/js/node_modules doesn't exist or is not a directory
      /home/mediacms.io/mediacms/frontend/src/static/node_modules doesn't exist or is not a directory
      /home/mediacms.io/mediacms/frontend/src/node_modules doesn't exist or is not a directory
      looking for modules in /home/mediacms.io/mediacms/frontend/node_modules
        single file module
          using description file: /home/mediacms.io/mediacms/frontend/package.json (relative path: ./node_modules/video.js)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/mediacms.io/mediacms/frontend/node_modules/video.js doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/mediacms.io/mediacms/frontend/node_modules/video.js.tsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/mediacms.io/mediacms/frontend/node_modules/video.js.ts doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/mediacms.io/mediacms/frontend/node_modules/video.js.jsx doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/mediacms.io/mediacms/frontend/node_modules/video.js.js doesn't exist
        /home/mediacms.io/mediacms/frontend/node_modules/video.js doesn't exist
      /home/mediacms.io/mediacms/node_modules doesn't exist or is not a directory
      /home/mediacms.io/node_modules doesn't exist or is not a directory
      /home/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
```

I am using Ubuntu Linux for what is worth. 


1. Can you checkout your branch to a new directory, make it work, and write the instructions here?
2. Can you also write to the documentation, how to use this, and include a screenshot too? 

Thanks!

**Comment by MrPercheul** [comment]
Hi, the issue should be fixed, there was a missing reference to video.js in the package json file

**Comment by MrPercheul** [comment]
Hi,
> I'd like to try this out locally, can you give detailed instructions on how to get it running?
Yes, I tried replicating the issue with a fresh install, and stumbed on the same thing. after a few changes ( 6f389c2 & 9488a86) It should be fixed (and I did "docker-compose up --force-recreate" to ensure all package changes are included)

>I see you are changing the version of videojs
Yep, corrected that to use the same as the rest.

>This also needs some mention on the documentation
Coming in the next commit =)

**Comment by MrPercheul** [comment]
HI @mgogoulos,

All should be included, please let me know if there are any issues left to run the project =)

**Comment by Micrufun** [comment]
@MrPercheul Thanks for your work :slightly_smiling_face: 

As far as I understand, the front-end is using a customized `mediacms-player` package inside `frontend/packages/player/package.json`. It's using `"video.js": "^7.12.3"` as a `peerDependencies`.

I see that this PR is using `video.js` as a direct dependency and by-passing the customized `mediacms-player`. I wonder if it would have any side-effect?

File: docker-compose-dev.yaml [hunk 1]
```diff
@@ -4,11 +4,11 @@ services:
   frontend:
     image: node:14
     volumes:
-      - ${PWD}/frontend:/home/mediacms.io/mediacms/frontend/
+      - ./frontend:/home/mediacms.io/mediacms/frontend/
```
Comment (mgogoulos) [inline docker-compose-dev.yaml:7]: why are you changing this?

File: docker-compose-dev.yaml [hunk 2]
```diff
@@ -4,11 +4,11 @@ services:
...
     working_dir: /home/mediacms.io/mediacms/frontend/
     command: bash -c "npm install && npm run start"
     env_file:
-      - ${PWD}/frontend/.env
+      - ./frontend/.env
```
Comment (mgogoulos) [inline docker-compose-dev.yaml:11]: same here

File: frontend/package.json [hunk 3]
```diff
@@ -37,6 +37,7 @@
     "ts-loader": "^9.2.3",
     "typescript": "^4.3.2",
     "url-loader": "^4.1.1",
+    "video.js": "^7.12.3",
```
Comment (mgogoulos) [inline frontend/package.json:40]: why are you adding this here?

**Review by mgogoulos — CHANGES_REQUESTED** [review]

**Comment by mgogoulos** [comment]
This PR is still needing work before it can be merged. I tested it locally and noticed the following:

1. when I try to post a comment it says 'comment submission has failed' but the comment succeeds in getting uploaded. I try as a self-registered user. I believe this behaviour is  not part of the main MediaCMS but it is getting introduced here ( i could be wrong). Please test and make sure what is the case here?
2. I am adding a few comments and then load the media page. Althought the comments API response arrives, it takes a few seconds to show the comments (initially it says 0 comments and then it shows the comments). This is not the existing behaviour 
3. Not sure about video.js and the comment @Micrufun has already made. Same for the PWD on docker-compose-dev.yml
Since there is not adequate testing already in place, I hesitate to merge anything that can potentially break existing behavior. Please make sure as few changes are there.

Thanks

File: docker-compose-dev.yaml (hunk 1 above)
Comment (MrPercheul) [reply docker-compose-dev.yaml:7]: Mistake from my side, I'll revert it!

File: docker-compose-dev.yaml (hunk 2 above)
Comment (MrPercheul) [reply docker-compose-dev.yaml:11]: Same, to be reverted

**Comment by MrPercheul** [comment]
Hi @mgogoulos,
I believe the changes should remove video,js as a direct dependency, and instead use `mediacms-player` as noted by micrufun. Also PWD has been put back (it was an error from my side as I am running things in Sudo).

However, I could not reproduce the "comment submission has failed" error, as for my version, I do not trigger it, and I encounter issue setting up a fresh install on a new VM

I get the following error running `docker-compose -f docker-compose-dev.yaml up` on my branch AND main

```
frontend_1          | > mediacms-frontend@0.9.1 start /home/mediacms.io/mediacms/frontend
frontend_1          | > mediacms-scripts development --config=./config/mediacms.config.js --host=0.0.0.0 --port=8088
frontend_1          | 
frontend_1          | sh: 1: mediacms-scripts: not found
frontend_1          | npm ERR! code ELIFECYCLE
frontend_1          | npm ERR! syscall spawn
frontend_1          | npm ERR! file sh
frontend_1          | npm ERR! errno ENOENT
frontend_1          | npm ERR! mediacms-frontend@0.9.1 start: `mediacms-scripts development --config=./config/mediacms.config.js --host=0.0.0.0 --port=8088`
frontend_1          | npm ERR! spawn ENOENT
frontend_1          | npm ERR! 
frontend_1          | npm ERR! Failed at the mediacms-frontend@0.9.1 start script.
frontend_1          | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

If you could be of guidance, as I do not know enough to fix/prevent this, it would be greatly appreciated

**Comment by sankasan** [comment]
@mgogoulos is this latest pull request now good to be upstreamed? We'd love to use this feature!

**Comment by MrPercheul** [comment]
Hi @mgogoulos,

Thank you for the pointers, I could not pinpoin precisely the issues, but going throught the suggested checks manage to unstuck me. 
I did the requested changes, Package-lock is now the same as Main and the comment submission issue is fixed! (I my lack of React experience, I noticed that I updated the comment list only once the video was loaded, and not at each event!)

Hope no other error is present!

**Comment by sankasan** [comment]
Great @MrPercheul this looks now ready for mainlining! @mgogoulos could you merge this?

**Comment by mgogoulos** [comment]
> Great @MrPercheul this looks now ready for mainlining! @mgogoulos could you merge this?

I'm on it, going to complete my testing on this tomorrow!

**Pull request merged**